server/core/src/https/mod.rs

```diff
@@ -253,6 +255,13 @@ pub async fn create_https_server(
 
     let trust_x_forward_for = config.trust_x_forward_for;
 
+    let cors_allowed_origins: Option<Vec<HeaderValue>> = config
+        .cors_allowed_origins
+        .as_ref()
+        .map(|os| os.into_iter().map(|o| o.parse().ok()).collect())
+        .flatten()
+        .and_then(|hvs: Vec<HeaderValue>| if hvs.is_empty() { None } else { Some(hvs) });
```

Comment:
Since it starts and ends as an option you should be able to avoid the re-wrapping-with-Some by dropping the flatten, I think?

Reply:
It's Option<Option<Vec>> initially, the last and_then converts Some(empty_vec) into None, otherwise the Option<Vec> remains Option<Vec>